### PR TITLE
build(deps): nightly toolchain update, nix fixes, CI test

### PR DESCRIPTION
**Pull request by rvolosatovs**
- Fix Nix development environment
- Run `cargo build` via Nix on CI to test the development environment
- Update rust nightly
- Update Nix flake

Switched to using a custom Nix Rust tooling fork until https://github.com/nix-community/fenix/issues/67 is fixed (https://github.com/nix-community/fenix/pull/68 PR filed)

### CODE REVIEW

**Comment by haraldh** [comment]
If you run cargo test, you want a kvm runner
```
Error: No supported backend found
```

**Comment by haraldh** [comment]
`runs-on: [self-hosted, linux]`

**Review by veehaitch — COMMENTED** [review]
Maybe also get rid of the currently unused `flake-compat` input?

File: flake.nix [hunk 1]
```diff
@@ -6,20 +6,17 @@
...
 
   outputs = { self, nixpkgs, fenix, flake-utils, ... }:
     # NOTE: musl is only supported on Linux.
     with flake-utils.lib; eachSystem [ system.x86_64-linux ] (system:
       let
-        pkgs = import nixpkgs {
-          inherit system;
-          overlays = [ ];
-        };
+        pkgs = import nixpkgs { inherit system; };
```
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply flake.nix]: That's an artifact left from usage of https://github.com/oxalica/rust-overlay

**Comment by rvolosatovs** [comment]
> Maybe also get rid of the currently unused `flake-compat` input?

It's used in https://github.com/enarx/enarx/blob/9402beeac7e939d97a2af24a416e4be1fca8d460/shell.nix#L1-L12

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -83,3 +83,11 @@ jobs:
...
           - name: release
             flag: --release
+
+  nix:
+    runs-on: [ self-hosted, linux ]
+    steps:
+    - uses: actions/checkout@v2.4.0
+    - uses: cachix/install-nix-action@v16
+    - run: nix flake check
+    - run: nix develop -c cargo test --no-default-features --features=backend-kvm --features=wasmldr
```
Comment (veehaitch) [inline .github/workflows/test.yml:93]: Maybe consider `--ignore-environment` for a less impure shell environment:

```suggestion
    - run: nix develop --ignore-environment -c cargo test --no-default-features --features=backend-kvm --features=wasmldr
```

**Review by haraldh — APPROVED** [review]

**Pull request merged**